packages/core/core/src/providers/registries.ts

```diff
@@ -0,0 +1,35 @@
+import { hooks } from '@strapi/utils';
+
+import { defineProvider } from './provider';
+import * as registries from '../registries';
+import { loadApplicationContext } from '../loaders';
+import * as draftAndPublishSync from '../migrations/draft-publish';
+
+export default defineProvider({
+  init(strapi) {
+    strapi
+      .add('content-types', () => registries.contentTypes())
+      .add('components', () => registries.components())
+      .add('services', () => registries.services(strapi))
+      .add('policies', () => registries.policies())
```

Comment:
stupid question alert, 
does it make sense to keep the concept of a registry and a provider here? What makes content types require a registry but not crons?

Reply:
Not sure I understand 🤔  
Crons have a registry but it's inside the cron service. and not in a shared registries folder.

Ideally I want to make those registries a bit smarter and move them into servies (like crons) 
Just don't want to stall too much that PR